### PR TITLE
[Ansible] Job uses GIT module to clone roles

### PR DESCRIPTION
**Pull request by UXabre**
Currently a git command is being executed, which causes the clone to fail if the roles folder is not empty; however, when running this as a recurring job, this fails after the first attempt.

By using the built-in GIT module, this problem seems to be fixed

### CODE REVIEW

**Comment by ares** [comment]
I suppose the force=yes does the trick, though is that safe/expected behavior? doesn't it override existing data?

this is what documentation says 

```
If yes, any modified files in the working repository will be discarded. Prior to 0.7, this was always 'yes' and could not be disabled. Prior to 1.9, the default was `yes`
```

I suppose there's a good reason they switched it to no.

**Comment by ares** [comment]
perhaps we could make this a template input to give user an option?

**Comment by UXabre** [comment]
Hey @ares, actually a good point, safest would be to actually keep the default of no.
I'll add this as template input but default to no. (but after reading you comment, I completely loathe my original decision to set force to "yes")

File: job_templates/ansible_roles_-_install_from_git.erb
Comment (ares) [inline job_templates/ansible_roles_-_install_from_git.erb:22]: Probably would be also good to specify options yes,no. I think if you set no as first, it will be default value

File: job_templates/ansible_roles_-_install_from_git.erb
Comment (ares) [inline job_templates/ansible_roles_-_install_from_git.erb:34]: We could just paste input('force') here if you decide to specify options, see above

**Review by ares — COMMENTED** [review]
One more suggestion, but looks better now :-)

File: job_templates/ansible_roles_-_install_from_git.erb
Comment (UXabre) [reply job_templates/ansible_roles_-_install_from_git.erb:22]: ah yes, forgot those values. However, from looking in the UI, there doesn't seem to be a way to set a default value (which is a shame :-( ), this is what caused me to add a fallback value to the input('force') below.

But perhaps, there's a way to set a default out of my knowledge?

Comment (ares) [reply job_templates/ansible_roles_-_install_from_git.erb:22]: There's no way to set default and yes, it's a shame :-) the only reasonable option is to specify options, with first being the preselected one. I moved the [tracking issue](https://projects.theforeman.org/issues/11367) from REX to core, it shouldn't be too complicated to add.

Comment (UXabre) [reply job_templates/ansible_roles_-_install_from_git.erb:22]: Hey @ares! Great, will look into that one first and come back to this once merged, if that is okay for you?

Comment (ares) [reply job_templates/ansible_roles_-_install_from_git.erb:22]: That would be awesome, feel free to ping me in that PR directly :-)

Comment (ares) [reply job_templates/ansible_roles_-_install_from_git.erb:22]: @UXabre, the default value was added in Foreman 1.24, I think this was the last thing to finish this PR, could you please add that? since `input('force')` will now always contain the value, the comment below still holds

**Review by ares — CHANGES_REQUESTED** [review]
switching to waiting on contributor, however very little is missing now to get this in, it's no longer blocked on core functionality

File: job_templates/ansible_roles_-_install_from_git.erb
Comment (UXabre) [reply job_templates/ansible_roles_-_install_from_git.erb:22]: Allright! Let me read up on this default value and finish this template! Thanks for following up!

**Comment by tbrisker** [comment]
On behalf of the Foreman team, thank you for your contribution to the Foreman community templates repository! 
This repository is being archived and all templates are migrated to the relevant repositories. If you wish to pursue this change, please re-open the PR against the relevant repository. For more information and background, please [click here]( https://community.theforeman.org/t/heads-up-community-templates-repo-will-be-archived-tomorrow/19539).

**Pull request closed without merging**
